<span style="font-size:16px">| BCOR2030, Fall 2023</span>
<br>
<span style="font-size:45px"><b>Confidence Interval</b></span><img style="float: right;padding-right:50px;" src="https://eccles.utah.edu/wp-content/themes/wptheme-avada5/logos/eccles_logo_footer.png" width="300">
<br>
<br>
<span style="font-size:18px; float:left;"><b>By Prof. Byoung-gyu Gong     </b></span>
<hr>
<br>
<br>
<br>
<span style="font-size:16px">Random sampling in python can be easily achieved by using 'pandas' library.</span>

<b>pip install numpy</b> - <span style="color:red"><i>High performing number manipulation library</i></span>
<br>
<b>pip install pandas</b> - <span style="color:red"><i>Data table manipulation library</i></span>
<br>
<b>pip install scipy</b> - <span style="color:red"><i>Statistics library</i></span>

In [2]:
import numpy as np
import pandas as pd
import scipy.stats as st

## 1. How to load your data using pandas
<hr style="border-top: 0.1px dotted navy" />
<span style="font-size:16px">To load your data set in the csv format, you can use 'read_csv()' function in the pandas library. Before loading your data set, you should drag and drop the data file into your vscode project folder. Unless you move it to the project folder, the function will not be able to locate where your file is.</span>

In [2]:
# Load your data set
df=pd.read_csv('EAI.csv')

Manager  Annual Salary
count  2500.00000    2500.000000
mean   1250.50000   51800.000000
std     721.83216    4000.000067
min       1.00000   40990.300000
25%     625.75000   49036.500000
50%    1250.50000   51837.300000
75%    1875.25000   54577.675000
max    2500.00000   63538.700000

## 2. Explore your data set.
<hr style="border-top: 0.1px dotted navy" />
<span style="font-size:16px">Functions such as <b>'type()'</b> and <b>'describe()'</b> can give you a brief and easy summary look of your data.</span>

In [ ]:
df.describe()
# Population parameter(average annual salary) is 51800.


## 3. Random Sampling in pandas
<hr style="border-top: 0.1px dotted navy" />
<span style="font-size:16px">By using <b>'sample()'</b> function, we can randomly select specific number of rows or observations in the data table. The 'sample()' function does not select the same rows in default setting, 'replace=False'. You can specify the number of samples you'd like to select by number or fraction. </span>

In [5]:
# Random sampling
n=50
df_sampl=df.sample(n)
df_sampl.describe()
# Your sample statistics(average of annual salary) is 52587.

Manager  Annual Salary
count    50.000000      50.000000
mean   1370.500000   52587.472000
std     636.541524    4472.632942
min      24.000000   41187.400000
25%     882.500000   49604.475000
50%    1376.500000   52013.450000
75%    1883.750000   54489.275000
max    2407.000000   62281.100000

## 4. Set Your Input Values to Calculate CI.
<hr style="border-top: 0.1px dotted navy" />

In [7]:
# Create key input indicators
## sample mean
m=df_sampl["Annual Salary"].mean()
## sample standard deviation
sd=df_sampl["Annual Salary"].std()
## confidence level
cf=0.95
## your degree of freedom(sample size)
df_n=n-1

print('sample mean=',m,'\n'
      'sample standard deviation=',sd,'\n'
      'your confidence level=',cf,'\n'
      'sample size(n-1)=',df_n)

sample mean= 52587.472 
sample standard deviation= 4472.63294232279 
your confidence level= 0.95 
sample size(n-1)= 49


## 6. Calculate T value
<hr style="border-top: 0.1px dotted navy" />

In [8]:
# T value

# First calculate t value given your confidence level
# abs: absolute value
# ppf: calculate cumulative distribution
# 1-cf/2: left and right tail of the t distribution

t_crit=np.abs(st.t.ppf((1-cf)/2, n-1))
print(t_crit)

2.009575234489209


## 7. Calculate Your Margin of Error (Interval Estimation)
<hr style="border-top: 0.1px dotted navy" />

In [9]:
# margin of error
(m-sd*t_crit/np.sqrt(n), m+sd*t_crit/np.sqrt(n))

(51316.36378367514, 53858.580216324866)

## 8. Interpret Your Output
<hr style="border-top: 0.1px dotted navy" />

Now, I am 95% sure that the population mean(average annual salary of the managers) is located inside the range of 51316 and 53858.

In [5]:
n=10
m=6
sd=0.5
cf=0.95
degf=n-1

t_crit=np.abs(st.t.ppf((1-cf)/2, n-1))
print(t_crit)
(m-sd*t_crit/np.sqrt(n), m+sd*t_crit/np.sqrt(n))

2.262157162740991


(5.642321547023714, 6.357678452976286)